transcom/mymove
# B 18411 add dest sit info on shipment dest address review screen

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A882385&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

Now that we are transitioning to the shipment's address being the "king" when referring to destination SIT service items' addresses, we need to alert the TOO when a shipment has destination SIT service items following a destination address change that impacts the shipment's status.

This PR does the following:
- added an alert to the `ShipmentAddressUpdateReviewRequestModal.jsx` component that displays when a shipment contains destination SIT service items
- updated existing tests to reflect changes

**NOTE** this is currently in waiting for approval of info banner language from the gov

### How to test

1. Access MM office and start up client/server
2. Request a destination address change to a shipment via [this endpoint](http://officelocal:3000/swagger-ui/prime.html#/mtoShipment/createNonSITAddressUpdateRequest)
3. Log in as a TOO and find the shipment
4. Review the shipment's changed address/request
5. If the shipment contains destination SIT service items, you should see an info alert box
6. If it does NOT contain destination SIT service items, it should not display
7. See screenshots for more info

## Screenshots

First we should request an update to the shipment destination address as the prime - this currently doesn't exist in the prime sim, so I'd recommend using your [local Swagger API](http://officelocal:3000/swagger-ui/prime.html#/mtoShipment/createNonSITAddressUpdateRequest) when you get your server & client running. Need to ensure the zip code changes enough to satisfy the criteria for the shipment's status to change
![Screenshot 2023-12-29 at 1 44 02 PM](https://github.com/transcom/mymove/assets/136510600/f46043f3-0338-4393-93e9-65d27a980dc1)

Once you do that, you should see this in the shipment box on the Move Details page as a TOO
![Screenshot 2023-12-29 at 1 34 57 PM](https://github.com/transcom/mymove/assets/136510600/c2c7738d-35ef-4932-996e-6f52fcd00fd8)

You can check out the review modal here after clicking "Edit Shipment"
![Screenshot 2023-12-29 at 1 35 10 PM](https://github.com/transcom/mymove/assets/136510600/ff005399-4fc1-418e-8c13-1b6ddad54e25)

Since the shipment doesn't have any destination SIT service items you won't see the alert
![Screenshot 2023-12-29 at 1 35 36 PM](https://github.com/transcom/mymove/assets/136510600/9ff95422-a9c6-4538-9c38-40e6a84501ea)

So let's add some service items to a shipment via prime sim
![Screenshot 2023-12-29 at 9 24 55 AM](https://github.com/transcom/mymove/assets/136510600/a4d3b3a0-76a3-43a7-83a4-5c2c04f50c5c)

Fill out the destination SIT form and send it, this creates four service items
![Screenshot 2023-12-29 at 9 27 39 AM](https://github.com/transcom/mymove/assets/136510600/00d888d2-9a94-4358-a83e-bfda741d4df9)

Log in as a TOO and find your shipment that is in a pending status and navigate to the review screen we saw earlier. You should now see the info alert letting the TOO know there are destination SIT service items in this shipment
![Screenshot 2023-12-29 at 1 37 06 PM](https://github.com/transcom/mymove/assets/136510600/c48e37a4-8777-4c71-852f-a5842582f7ff)
